File: RedBusApp/src/app/models/class/Bus.Model..ts
import { Bus } from "./Bus";

export class BusModel extends Bus {
    fromLocationName: string;
    toLocationName: string;
    seatNos: string[];

    constructor() {
        super(); // calls Bus constructor
        // any extra initialization goes here
        this.fromLocationName = "";
        this.toLocationName = "";
        this.seatNos = [];
    }
}

File: APIProject/APIProject/Models/BusModel.cs
using System.ComponentModel.DataAnnotations;

namespace APIProject.Models
{
    public class BusModel: Bus
    {
        public string FromLocationName { get; set; } = String.Empty;
        public string ToLocationName { get; set; } = String.Empty;
        public string[] SeatNos { get; set; } = [];
    }
}